Lib/unittest/mock.py

```diff
@@ -2816,6 +2819,59 @@ def __set__(self, obj, val):
         self(val)
 
 
+class ThreadingMock(MagicMock):
```

Comment:
My discussion with @voidspace was to use Mixins instead of directly inheriting `MagicMock`.

Reply:
I can change this to have a `ThreadingMixin` and then have `ThreadingMock` to inherit from `MagicMock` and `ThreadingMixing`.

@tirkarthi @lisroach @voidspace do you prefer that?

Reply:
@mariocj89 sorry for such a late reply, but yes I agree with using mixins to keep it consistent with the rest of the module

Reply:
@mariocj89 - I'm confused, I thought I saw a commit where you made this change? Sounds like it should happen before we land this...